front/lib/api/assistant/onboarding.ts

```diff
@@ -177,25 +190,82 @@ function buildOnboardingPrompt(options: {
     }
   }
 
+  // Build suggested tool names for reference
+  const suggestedTopToolNames = topTools
+    .map((sId) => asDisplayName(sId))
+    .join(", ");
+
+  // Build toolSetup directives for "connect more tools" flow
+  const otherTools = tools.slice(2);
+  const otherToolSetupDirectives = otherTools
+    .map((sId) => `:toolSetup[Connect ${asDisplayName(sId)}]{sId=${sId}}`)
+    .join(" ");
+
+  // Build the first message section based on whether a tool is already configured
+  const firstMessageSection = alreadyConfiguredTopTool
+    ? buildFirstMessageWithConfiguredTool(alreadyConfiguredTopTool)
+    : buildFirstMessageWithToolSetup(
+        topToolsWithDescriptions,
+        topToolSetupDirectives,
+        suggestedTopToolNames
+      );
+
   return `<dust_system>
 You are onboarding a brand-new user to Dust.
 ${userContext}
 
 ## CRITICAL RULES
 
 1. EVERY message MUST end with at least one interactive element (toolSetup or quickReply)
-2. Keep messages SHORT - 3-4 lines maximum for the first message
+2. Keep messages SHORT - 3-4 lines maximum
 3. Be direct and action-focused - get to the tool connection quickly
 
-## CRITICAL: Do not hallucinate use cases
-
-- NEVER suggest cross-tool queries like "get answers from multiple sources" or "search across X and Y"
-- NEVER assume what data the user has (no "find your design docs", "search your specs", "find feedback on the checkout redesign", etc.)
+### Do not hallucinate
```

Comment:
Same here, why remove the critical? Cause it did hallucinate use cases that's why Henry upper-cased a CRITICAL 😄